Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -122,14 +122,21 @@ public IDisposable Consume(Action<IConsumeConfiguration> configure)
                         x.Item3.ConsumerTag,
                         x.Item3.IsExclusive,
                         x.Item3.Arguments,
-                        (body, properties, receivedInfo, cancellationToken) =>
+                        async (body, properties, receivedInfo, cancellationToken) =>
                         {
                             var rawMessage = produceConsumeInterceptor.OnConsume(
                                 new ConsumedMessage(receivedInfo, properties, body)
                             );
-                            return x.Item2(
-                                rawMessage.Body, rawMessage.Properties, rawMessage.ReceivedInfo, cancellationToken
-                            );
+                            try
+                            {
+                                return await x.Item2(
+                                    rawMessage.Body, rawMessage.Properties, rawMessage.ReceivedInfo, cancellationToken
+                                ).ConfigureAwait(false);
```

Comment:
`ConfigureAwait(false)` is required here since `SynchronizationContext.Current` is `Xunit.Sdk.MaxConcurrencySyncContext` with 6 custom worker threads rather than threads from the thread pool. I spent some time to figure out why part of the tests began to give a timeout error.